books/xdoc/fancy/xdoc.js

```diff
@@ -1051,184 +1051,131 @@ document.addEventListener("DOMContentLoaded", () => {
-	    return matches.sort(compare);
-	}
-
-	// Lots of elements -- do something fancier for more speed.  Since we
-	// are only going to show 20 results, it is definitely safe to throw
-	// away any entries whose UID is larger than some CUTOFF, as long as
-	// CUTOFF is bigger than at least 20 UIDs.  So, gather a reasonable
-	// subset of UIDs, sort them, and pick the 20th one.  Then throw away
-	// everything past it in a linear sweep.
-	var first_uids = [];
-	for(var i = 0; i < 100; ++i) {
-	    first_uids.push(matches[i].uid);
-	}
-	//console.log("Gathered up UIDs: ", first_uids);
-	first_uids = first_uids.sort(function(a,b){ return a-b; });
-	//console.log("Sorted UIDs:", first_uids);
-	var cutoff = first_uids[19];
-
-	var consider = [];
-	for(var i = 0;i < matches.length; ++i) {
-	    var entry = matches[i];
-	    if (entry.nicelow == curr || entry.uid < cutoff) {
-		consider.push(entry);
-	    }
-	}
-	//console.log("Down to " + consider.length + " entries to consider:", consider);
-
-	// Then we just sort whatever survived using our ordinary comparison
-	// function, which now doesn't need to consider the vast majority of
-	// the array.
-	return consider.sort(compare);
-    };
-
-    engine1.initialize();
-
-    $("#jump").typeahead({
-			     highlight: true,
-			     hint: true,
-			     autoselect: true
-			 },
-			 {
-			     name: "topics",
-			     displayKey: 'nicename',
-			     source: engine1.ttAdapter(),
-			     templates:
-			     {
-				 suggestion: jumpRender
-			     }
-			 });
+    // Custom substring matcher for infix matching with ranking and ACL2 Sources priority
+    function substringMatcher(data) {
+        return function findMatches(q, cb) {
+            var matches = [];
+            var qlc = q.toLowerCase();
+            for (var i = 0; i < data.length; i++) {
+                var name = data[i].nicename.toLowerCase();
+                var value = data[i].value.toLowerCase();
+                var rank = null;
+                if (name === qlc || value === qlc) {
+                    rank = 0; // exact match
+                } else if (name.startsWith(qlc) || value.startsWith(qlc)) {
+                    rank = 1; // prefix match
+                } else if (name.indexOf(qlc) !== -1 || value.indexOf(qlc) !== -1) {
+                    rank = 2; // substring match
+                }
+                if (rank !== null) {
+                    matches.push({ ...data[i], _rank: rank });
+                }
+            }
+            // Sort by rank, then ACL2 Sources, then nicename
+            matches.sort(function(a, b) {
+                if (a._rank !== b._rank) return a._rank - b._rank;
+
+                // ACL 2 Package Priority
+                //var pkgA = xdataObj.topicBasepkg(a.value);
+                //var pkgB = xdataObj.topicBasepkg(b.value);
+                //if (pkgA === 'ACL2' && pkgB !== 'ACL2') return -1;
+                //if (pkgA !== 'ACL2' && pkgB === 'ACL2') return 1;
+
```

Comment:
I'd remove this commented-out code (unless there is a reason I'm not seeing to keep it in).